### DISCUSSION TITLE
Can i show the Date Time in the sharepoint local time zone

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [microsoft-search/pnp-modern-search#3926] [Q&A]
**I am working on a SharePoint online site collection which have this regional settings:-**

![image](https://github.com/user-attachments/assets/06d6d17d-a0a7-46a8-904c-68cfdcc78e8f)

**And i have a SharePoint column of type Date/Time, now inside SharePoint list view the date/time will be show as follow:-**

![image](https://github.com/user-attachments/assets/225198b2-3e61-40e3-9173-c979747e9ad9)

**which respect the site regional settings. now for the above site column >> i have a managed property and when i am showing this manage property inside the Search Result web part i will get the value based on UTC, as follow:-**

![image](https://github.com/user-attachments/assets/7a729981-79aa-4e7d-be5d-e3e0eaf6523f)

**so how i can show the date/time inside my Search Result to be equal to what is shown inside the list view?**

**Here is how i am rendering the column value inside the Search Result web part settings (resource.fields.RefinableDate01):-**

![image](https://github.com/user-attachments/assets/91ab5b51-586f-470c-aef1-6c7b1c957692)

**Comment by kasperbolarsen** [comment]
This one might be helpful, https://spdcp.com/2019/12/21/local-time-zones-for-pnp-modern-search-web-part/

**Reply by mvcsharepointdev** [reply]
@kasperbolarsen thanks for the reply, now i tried this formula:-

`{{ getDate (getDate (slot item @root.slots.PUETA "YYYY-MM-DDTHH:mm:ss.0000000\Z") "ddd, MMM DD h:mm a" 3 }}`

Where i defined 3 to get the time zone of the current web, as per the documentation:-

![image](https://github.com/user-attachments/assets/d15d78af-500a-4bcc-9b08-03e5c9322bfe)


but still there are differences, here are the actual date/time inside the list view:-

![image](https://github.com/user-attachments/assets/e36bc7f6-86df-4dd7-862d-7b38bad41e40)


and here is how they are shown using the above expression/formula inside the PnP Modern search web part:-

![image](https://github.com/user-attachments/assets/36590553-4582-4f9e-b220-d8c634e23f11)


so still there are those differences: -

21 August 2024 11 PM  >>>>>>>>  22 August 2024 1 am
21 August 2023 10 PM  >>>>>>>>  22 August 2023 12 am


so the gap get minimized, but still there are 2 hours differences .. any advice? can this be fixed?
